pkg/apis/core/validation/validation.go

```diff
@@ -2689,6 +2711,20 @@ func validateContainers(containers []core.Container, isInitContainers bool, volu
 	return allErrs
 }
 
+//check that we have at least one container in the pod which is not set to lifecyle type sidecar
+func validateSidecars(containers []core.Container, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	for _, container := range containers {
+		if container.Lifecycle == nil || container.Lifecycle.Type == nil || *container.Lifecycle.Type != core.LifecycleTypeSidecar {
+			//if we find a container that is not a sidecar, return
+			return allErrors
+		}
+	}
+
+	return append(allErrors, field.Forbidden(fldPath, fmt.Sprint("must not only have containers with lifecyle type == sidecar in a pod")))
```

Comment:
"must have at least one container with lifecycle type != sidecar"

Reply:
or "... with a standard lifecycle"

Reply:
yeah "must have at least one container with lifecycle type != sidecar" I think works better, specifying "with a standard lifecycle" may be confusing as this field is optional, so nil is fine